Bug with reading in Knotvectors

I am currently trying to read in the following file

[knotvectors.txt](https://github.com/user-attachments/files/20242500/knotvectors.txt)

and i am failing with 

```
>           if not line.strip():
E           AttributeError: 'float' object has no attribute 'strip'

../../02_applications/miniconda3/envs/meshpy/lib/python3.12/site-packages/fourcipp/legacy_io/knotvectors.py:60: AttributeError
```

The error happens at line 18 (of the provided file) in the Section, i.e. the first 0.0

--- a/src/fourcipp/legacy_io/knotvectors.py
+++ b/src/fourcipp/legacy_io/knotvectors.py
@@ -54,11 +54,18 @@ def read_knotvectors(list_of_lines):
     while list_of_lines:
         line = list_of_lines.pop(0)
 
-        # Empty line
-        if not line.strip():
+        if isinstance(line, str):
+            if not line.strip():
+                # Skip empty line
+                continue
+            else:
+                line_list = line.split()
+        elif isinstance(line, (int, float)):
+            line_list = [line]
+        elif line is None:
             continue
-
-        line_list = line.split()
+        else:
+            raise TypeError(f"Error while parsing knotvectors line: {line}!")
 
         # Key value case
         if len(line_list) == 2:

--- a/src/fourcipp/legacy_io/knotvectors.py
+++ b/src/fourcipp/legacy_io/knotvectors.py
@@ -51,6 +51,7 @@ def read_knotvectors(list_of_lines):
     knots_data = {}
     patches = []
 
+    latest_nurb_dimension = None
     while list_of_lines:
         line = list_of_lines.pop(0)
 
@@ -77,9 +78,11 @@ def read_knotvectors(list_of_lines):
 
             # End reading in patch
             elif key == "END":
+                if "NURBS_DIMENSION" in patch_data:
+                    latest_nurb_dimension = patch_data.pop("NURBS_DIMENSION")
                 # Check dimension
                 if (nurbs_dimension := len(patch_data["knot_vectors"])) != (
-                    nurbs_dimension_expected := patch_data.pop("NURBS_DIMENSION")
+                    nurbs_dimension_expected := latest_nurb_dimension
                 ):
                     raise ValueError(
                         f"Expected {nurbs_dimension_expected} knot vectors, got {nurbs_dimension}"

--- a/tests/fourcipp/legacy_io/test_knotvectors.py
+++ b/tests/fourcipp/legacy_io/test_knotvectors.py
@@ -56,7 +56,6 @@ def fixture_reference_knotvector_lines():
         "1.0",
         "1.0",
         "END                                   NURBSPATCH",
-        "NURBS_DIMENSION                       2",
         "BEGIN                                 NURBSPATCH",
         "ID                                    2",
         "NUMKNOTS                              7",

--- a/tests/fourcipp/legacy_io/test_knotvectors.py
+++ b/tests/fourcipp/legacy_io/test_knotvectors.py
@@ -30,64 +30,81 @@ from fourcipp.legacy_io.knotvectors import read_knotvectors, write_knotvectors
 def fixture_reference_knotvector_lines():
     """Reference lines."""
     return [
-        "NURBS_DIMENSION                       2",
+        "NURBS_DIMENSION                       3",
         "BEGIN                                 NURBSPATCH",
         "ID                                    1",
-        "NUMKNOTS                              7",
-        "DEGREE                                2",
-        "TYPE                                  Interpolated",
-        "0.0",
-        "0.0",
-        "0.0",
-        "0.5",
-        "1.0",
-        "1.0",
-        "1.0",
-        "",
         "NUMKNOTS                              8",
         "DEGREE                                2",
         "TYPE                                  Interpolated",
-        "0.0",
-        "0.0",
-        "0.0",
-        "0.3333333333333333",
-        "0.6666666666666666",
-        "1.0",
-        "1.0",
-        "1.0",
+        "0.000000000000",
+        "0.000000000000",
+        "0.000000000000",
+        "0.333333333333",
+        "0.666666666667",
+        "1.000000000000",
+        "1.000000000000",
+        "1.000000000000",
+        "NUMKNOTS                              6",
+        "DEGREE                                2",
+        "TYPE                                  Interpolated",
+        "0.000000000000",
+        "0.000000000000",
+        "0.000000000000",
+        "1.000000000000",
+        "1.000000000000",
+        "1.000000000000",
+        "NUMKNOTS                              8",
+        "DEGREE                                2",
+        "TYPE                                  Interpolated",
+        "0.000000000000",
+        "0.000000000000",
+        "0.000000000000",
+        "0.333333333333",
+        "0.666666666667",
+        "1.000000000000",
+        "1.000000000000",
+        "1.000000000000",
         "END                                   NURBSPATCH",
+        "NURBS_DIMENSION                       2",
         "BEGIN                                 NURBSPATCH",
         "ID                                    2",
-        "NUMKNOTS                              7",
+        "NUMKNOTS                              9",
         "DEGREE                                2",
         "TYPE                                  Interpolated",
-        "0.0",
-        "0.0",
-        "0.0",
-        "0.5",
-        "1.0",
-        "1.0",
-        "1.0",
-        "NUMKNOTS                              8",
+        "0.000000000000",
+        "0.000000000000",
+        "0.000000000000",
+        "0.250000000000",
+        "0.500000000000",
+        "0.750000000000",
+        "1.000000000000",
+        "1.000000000000",
+        "1.000000000000",
+        "NUMKNOTS                              6",
         "DEGREE                                2",
         "TYPE                                  Interpolated",
-        "0.0",
-        "0.0",
-        "0.0",
-        "0.3333333333333333",
-        "0.6666666666666666",
-        "1.0",
-        "1.0",
-        "1.0",
+        "0.000000000000",
+        "0.000000000000",
+        "0.000000000000",
+        "1.000000000000",
+        "1.000000000000",
+        "1.000000000000",
         "END                                   NURBSPATCH",
     ]
 
 
 def test_knotvectors_read_and_write(reference_knotvector_lines):
     """Test read and write knotvectors."""
-    lines = write_knotvectors(read_knotvectors(reference_knotvector_lines))
-    for k in range(len(reference_knotvector_lines)):
-        assert reference_knotvector_lines[k].split() == lines[k].split()
+    # Copy the reference lines because read_knotvectors removes all lines
+    lines = write_knotvectors(read_knotvectors(reference_knotvector_lines.copy()))
+
+    for line, reference_line in zip(lines, reference_knotvector_lines):
+        if isinstance(line, str):
+            assert line.split() == reference_line.split()
+        elif isinstance(line, float):
+            assert line == float(reference_line)
+        else:
+            raise ValueError(f"Unexpected type {type(line)} in line: {line}")
 
 
 def test_wrong_knotvectors_dimension_mismatch():

